packages/dev/core/src/Materials/Node/Blocks/Input/prePassTextureBlock.ts

```diff
@@ -0,0 +1,180 @@
+        if (output === this._outputs[0]) {
+            return this._positionSamplerName;
+        }
+
+        if (output === this._outputs[1]) {
+            return this._depthSamplerName;
+        }
+
+        if (output === this._outputs[2]) {
+            return this._normalSamplerName;
+        }
+
+        return "";
+    }
+
+    /**
+     * Gets the position texture
+     */
+    public get position(): NodeMaterialConnectionPoint {
+        return this._outputs[0];
+    }
+
+    /**
+     * Gets the depth texture
+     */
+    public get depth(): NodeMaterialConnectionPoint {
+        return this._outputs[1];
+    }
+
+    /**
+     * Gets the normal texture
+     */
+    public get normal(): NodeMaterialConnectionPoint {
+        return this._outputs[2];
+    }
+
+    /**
+     * Gets the sampler name associated with this image source
+     */
+    public get positionSamplerName(): string {
+        return this._positionSamplerName;
+    }
+
+    /**
+     * Gets the sampler name associated with this image source
+     */
+    public get normalSamplerName(): string {
+        return this._normalSamplerName;
+    }
+    /**
+     * Gets the sampler name associated with this image source
+     */
+    public get depthSamplerName(): string {
+        return this._depthSamplerName;
+    }
+
+    /**
+     * Gets the current class name
+     * @returns the class name
+     */
+    public getClassName() {
+        return "PrePassTextureBlock";
+    }
+
+    protected _buildBlock(state: NodeMaterialBuildState) {
+        super._buildBlock(state);
+
+        if (state.target === NodeMaterialBlockTargets.Vertex) {
+            this._positionSamplerName = "prepassPositionSampler";
+            this._depthSamplerName = "prepassDepthSampler";
+            this._normalSamplerName = "prepassNormalSampler";
+
+            // Unique sampler names for every prepasstexture block
+            state.sharedData.variableNames.prepassPositionSampler = 0;
+            state.sharedData.variableNames.prepassDepthSampler = 0;
+            state.sharedData.variableNames.prepassNormalSampler = 0;
+
+            // Declarations
+            state.sharedData.blockingBlocks.push(this);
+            state.sharedData.textureBlocks.push(this);
+            state.sharedData.bindableBlocks.push(this);
+        }
+
+        state._emit2DSampler(this._positionSamplerName);
+        state._emit2DSampler(this._depthSamplerName);
+        state._emit2DSampler(this._normalSamplerName);
+
+        return this;
+    }
+
+    public bind(effect: Effect, nodeMaterial: NodeMaterial, mesh?: Mesh) {
+        if (!mesh) {
+            return;
+        }
+
+        const scene = mesh.getScene();
+        const prePassRenderer = scene.enablePrePassRenderer();
+        if (!prePassRenderer) {
+            return;
+        }
+
+        const sceneRT = prePassRenderer.defaultRT;
+        if (!sceneRT.textures) {
+            return;
+        }
+        prePassRenderer.defaultRT.level = 1;

```

Comment:
it should be removed as I added specific code in `textureBlock.ts`.

FYI, it was here because the situation is a bit of a hack : 

`level` is set here in `textureBlock` : https://github.com/BabylonJS/Babylon.js/blob/cbb72fc23b511ba64c45001ad829ba86cd7e98ec/packages/dev/core/src/Materials/Node/Blocks/Dual/textureBlock.ts#L386

`PrePassTextureBlock` being an `ImageSourceBlock` cannot set its own `texture` in `textureBlock`, because it has 3 outputs. Therefore [this getter](https://github.com/BabylonJS/Babylon.js/blob/cbb72fc23b511ba64c45001ad829ba86cd7e98ec/packages/dev/core/src/Materials/Node/Blocks/Dual/textureBlock.ts#L44C3-L44C3) will return `null`, and the `bind` will never set the necessary uniform `this._textureInfoName`